language/move-prover/src/bytecode_translator.rs

```diff
@@ -205,16 +205,24 @@ impl<'env> ModuleTranslator<'env> {
             );
         }
         let type_value = format!("StructType({}, {})", struct_name, field_types);
-        if struct_name == "LibraAccount_T" {
+        if struct_name == "$LibraAccount_T" {
```

Comment:
Is I said in IM, can we just have a constant in this module:

```
const FORWARD_DECLARED_TYPES : &[&str] = &["T1", "T2", ...];
````

Then check `FORWARD_DECLARED_TYPES.contains(&struct_name)`. This duplication is not needed.

Reply:
I am not sure if this would be possible in this case. `$LibraAccount_Balance` needs to generate/emit different code to `$LibraAccount_T` because `$LibraAccount_Balance` takes a type parameter, but `$LibraAccount_T` does not.

Reply:
Ok, SG. Let me know once you have brought back the errors, and I can approve.

Reply:
Sounds good.